src/OpenFeature.Contrib.Providers.Flagd/FlagdProvider.cs

```diff
@@ -539,30 +540,49 @@ private Service.ServiceClient BuildClientForPlatform(Uri url)
 
             if (!useUnixSocket)
             {
-#if NET462
+#if NET462_OR_GREATER
                 var handler = new WinHttpHandler();
 #else
                 var handler = new HttpClientHandler();
 #endif
                 if (_config.UseCertificate)
                 {
-#if NET5_0_OR_GREATER
-                    if (File.Exists(_config.CertificatePath)) {
+                    if (File.Exists(_config.CertificatePath))
+                    {
                         X509Certificate2 certificate = new X509Certificate2(_config.CertificatePath);
+#if NET5_0_OR_GREATER
                         handler.ServerCertificateCustomValidationCallback = (message, cert, chain, _) => {
                             // the the custom cert to the chain, Build returns a bool if valid.
                             chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
                             chain.ChainPolicy.CustomTrustStore.Add(certificate);
                             return chain.Build(cert);
                         };
-                    } else {
-                        throw new ArgumentException("Specified certificate cannot be found.");
-                    }
+#elif NET462_OR_GREATER
+                        handler.ServerCertificateValidationCallback = (message, cert, chain, errors) => {
+                            if ((errors & SslPolicyErrors.None) > 0) { return true; }
```

Comment:
`errors` is of type `SslPolicyErrors`, correct? So this is essentially a ANDing some value (0,1,2,4) with 0, so won't this never return true?

Maybe I'm missing something here, but I think this return will never execute.

Reply:
You are correct, had that fix in a stash I forgot about.